### PR TITLE
Fix codeclimate issues (except for interactive SDK)

### PR DESCRIPTION
**Pull request by tshastry**

### CODE REVIEW

File: testing/discovery_interface.py [hunk 1]
```diff
@@ -163,16 +156,19 @@ def reload_discovery_config():
     return ('result' in payload and payload['result'] == "success")
 
 
-def shutdown_discovery(shutdown=True):
+def shutdown_discovery(volume=None):
     """
     Shuts down the Discovery engine Docker container
     """
-    if not shutdown:
+    if str(os.environ.get('SHUTDOWN_DISCOVERY', True)) == "False":
```
Comment (ageojo) [inline testing/discovery_interface.py]: do we need to cast to a string?

File: testing/parse_tests.py [hunk 2]
```diff
@@ -75,22 +103,17 @@ def find_whitelists(test_file):
...
-            continue
-        if line.startswith("domain_whitelist"):
-            domain_whitelist = format_whitelist(line)
-            continue
-        # we only want to return whatever is leftover after the comments and
-        # whitelists are removed from the test_file
-        test_file = test_file[i:]
-        break
-        
+    intents = [i for i in test_file if i.startswith("intent_whitelist")]
```
Comment (ageojo) [inline testing/parse_tests.py]:
```suggestion
    intents = [intent for intent in test_file if intent.startswith("intent_whitelist")]
```

File: testing/parse_tests.py [hunk 3]
```diff
@@ -75,22 +103,17 @@ def find_whitelists(test_file):
...
+    domains = [d for d in test_file if d.startswith("domain_whitelist")]
```
Comment (ageojo) [inline testing/parse_tests.py]:
```suggestion
    domains = [domain for domain in test_file if domain.startswith("domain_whitelist")]
```

File: testing/parse_tests.py [hunk 4]
```diff
@@ -75,22 +103,17 @@ def find_whitelists(test_file):
...
+
+    intent_whitelist = format_whitelist(intents[0]) if intents else ["any"]
+    domain_whitelist = format_whitelist(domains[0]) if domains else ["any"]
+
+    test_file = [l for l in test_file if l not in intents + domains]
```
Comment (ageojo) [inline testing/parse_tests.py]:
```suggestion
    test_file = [line for line in test_file if not line in intents + domains]
```

File: testing/discovery_interface.py (hunk 1 above)
Comment (tshastry) [reply testing/discovery_interface.py]: To achieve boolean behavior, yes.

**Pull request merged**